recipes/sbepp/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.files import export_conandata_patches, get, copy, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "sbepp"
+    description = "C++ implementation of the FIX Simple Binary Encoding"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/OleksandrKvl/sbepp"
+    topics = ("trading", "fix", "sbe")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_sbeppc": [True, False],
+        "allow_clang_11": [None, True, False]
+    }
+    default_options = {
+        "with_sbeppc": True,
+        "allow_clang_11": None
```

Comment:
Instead of adding this option, I think this should be handled with `ConanInvalidConfiguration` when `self.settings.compiler.libcxx="libstdc++"` in the `validate()` method:

```
if self.settings.compiler== "clang" and Version(self.settings.compiler.version).major == "11" and \
    self.settings.compiler.libcxx == "libstdc++":
    raise ConanInvalidConfiguration(f"{self.ref} requires C++ filesystem library, not supported by libstdc++, please use libstdc++11 instead")
```

Reply:
This problem is specific only to Conan CI which uses very old libstdc++ version in its clang-11 image (see [here](https://github.com/conan-io/conan-center-index/pull/4630#discussion_r609051680)). It's still possible to build it locally using clang-11 and correct libstdc++ version installed so I see no reason to mark this configuration as invalid. My understanding is that your approach will completely disallow such a build.

Reply:
Hi @OleksandrKvl, thanks for the reference. I was investigating this a bit more and seems like indeed the libstdc++ implementation in the docker image in the CI is too old (until GLIBCXX_3.4.25). We are going to have a look at this because we should have changed this image some time ago and for some reason it was not done. I'll ping you when we check the possible options. Thanks a lot!

Reply:
Conan CI uses an old version of libstdc++ to keep compatibility and avoid breaking users. Once we start to build packages in CCI is almost impossible to update something due system libraries compatibility.

The problem is [filesystem](https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/src/filesystem/path.cc) uses `_GLIBCXX_USE_CXX11_ABI=1` by default, and it's the same behavior when using `compiler.libcxx=libstdc++11` and it works well with Clang 11, but for `libcxx=libstdc++` it does not work as you see.

Please, do not use options for settings, removed the clang 11 option and follow Carlos' approach. There are more recipes in CCI using the same restriction.

And to be honest, on Conan v2, the libstdc++11 is the default configuration, so users should not be so affected. I'll try to find some way to update the libstdc++ meanwhile, but it's hard to do it as it may break CCI as well.

Reply:
Got it, will follow suggested approach. Thanks!

Reply:
@OleksandrKvl Good news, we just removed Clang 11 configuration for Conan v1. So you can remove all this clang checking now, your recipe should pass without problem. 😄

Reply:
@uilianries that's great! I have another question: I already bumped project's version to 1.0.1 (in its repo, not here), can I just skip 1.0.0 here and use newer version?

Reply:
@OleksandrKvl of course! It's your PR, please, update it to 1.0.1 here 👍

Reply:
applied suggested changes